storage/gcp/gcp.go

```diff
@@ -302,6 +340,92 @@ func (s *spannerSequencer) assignEntries(ctx context.Context, entries [][]byte)
 	return uint64(next), nil
 }
 
+// consumeEntries calls f with previously sequenced entries.
+//
+// Once f returns without error, the entries it was called with are considered to have been consumed and are
+// removed from the Seq table.
+//
+// Returns true if some entries were consumed as a weak signal that there may be further entries waiting to be consumed.
+func (s *spannerSequencer) consumeEntries(ctx context.Context, limit uint64, f consumeFunc) (bool, error) {
```

Comment:
What if the logic retrieves `limit + 1` sequenced entries to provide a strong signal?

Reply:
At that point it may as well just do the work (and so on), but I'm trying to keep the chunks of work bounded. I figure that the cost of an "false positive" signal here is not really that great.

This is all open to change in the near future though, if it turns out to perform poorly.